usage/test-framework/src/main/java/org/apache/brooklyn/test/framework/TestFrameworkAssertions.java

```diff
@@ -39,122 +42,216 @@
+                                           String target,
+                                           final Supplier<T> actualSupplier) {
+
+        AssertionSupport support = new AssertionSupport();
+        checkAssertions(support, flags, assertions, target, actualSupplier);
+        support.validate();
+    }
+
+
+    public static <T> void checkAssertions(Map<String,?> flags,
+                                           List<Map<String, Object>> assertions,
+                                           String target,
+                                           final Supplier<T> actualSupplier) {
+
+        AssertionSupport support = new AssertionSupport();
+        for (Map<String, Object> assertionMap : assertions) {
+            checkAssertions(support, flags, assertionMap, target, actualSupplier);
         }
-        return Maybe.absent(String.format("No predicate found with signature [%s]", predicateKey));
+        support.validate();
     }
 
-    /**
-     * Builds a predicate that checks if a string contains the supplied value
-     *
-     * @param predicateTarget
-     * @return {@link Predicate}
-     */
-    private static Predicate<String> buildContainsPredicate(final String predicateTarget) {
-        return new Predicate<String>() {
+    public static <T> void checkAssertions(final AssertionSupport support,
+                                           Map<String,?> flags,
+                                           final List<Map<String, Object>> assertions,
+                                           final String target,
+                                           final Supplier<T> actualSupplier) {
 
-            @Override
-            public boolean apply(@Nullable final String input) {
-                return StringUtils.contains(input, predicateTarget);
-            }
+        for (Map<String, Object> assertionMap : assertions) {
+            checkAssertions(support, flags, assertionMap, target, actualSupplier);
+        }
+    }
 
-            @Override
-            public String toString() {
-                return String.format("TestFrameworkAssertions.contains(%s)", predicateTarget);
-            }
-        };
+    public static <T> void checkAssertions(final AssertionSupport support,
+                                           Map<String,?> flags,
+                                           final Map<String, Object> assertions,
+                                           final String target,
+                                           final Supplier<T> actualSupplier) {
+
+        if (null == assertions) {
+            return;
+        }
+        try {
+            Asserts.succeedsEventually(flags, new Runnable() {
+                @Override
+                public void run() {
+                    T actual = actualSupplier.get();
+                    checkActualAgainstAssertions(support, assertions, target, actual);
+                }
+            });
+        } catch (Throwable t) {
```

Comment:
I'm very wary of swallowing the wrong kind of exception - e.g. if it was an `InterruptedException`, then important the thread is left in an interrupted state (see http://www.ibm.com/developerworks/library/j-jtp05236/).

Perhaps do `support.fail(t); Exceptions.propagateIfFatal(t);`?

(but I haven't checked what support.fail does).